src/vampir/spec.lisp

```diff
@@ -202,3 +202,71 @@
 
 (defun make-tuples (&key wires)
   (make-instance 'tuple :wires wires))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Boolean
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defparameter *bool* 
+(let ((x-wire (make-wire :var :x)))
+  (make-alias :name :bool
+                  :inputs (list :x)
+                  :body
+                  (list
+                   (make-equality :lhs
+                                      (make-infix :op :*
+                                                      :lhs x-wire
+                                                      :rhs
+                                                      (make-infix
+                                                       :lhs x-wire
+                                                       :rhs (make-constant :const 1)
+                                                       :op :-))
+                                      :rhs (make-constant :const 0))
+                   x-wire))))
+
+(defun bool (x)
+          (make-application :func :bool
+                                :arguments
+                                (list x)))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Range inductive definition up to 32-bits
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+
+(defparameter *next-range*
+  (let ((a-wire (make-wire :var :a)))
+  (make-alias :name :next-range
+	      :inputs (list :range :a)
+	      :body
+	      (list (make-bind :names :a0
+			       :value
+			       (make-application :func :bool
+						 :arguments (list
+							     (make-application :func
+									       :fresh
+									       :arguments (make-infix :op :%
+												      :lhs a-wire
+												      :rhs (make-constant :const 2))))))
+		    (make-bind :names :a1
+			       :value
+			       (make-application :func :fresh
+						 :arguments (make-infix :op :\\
+									:lhs a-wire
+									:rhs (make-constant :const 2))))
+		    (make-equality :lhs
+				   a-wire
+				   :rhs
+				   (make-infix :op :+
+					       :lhs (make-wire :var :a0)
+					       :rhs (make-infix :op :*
+								:lhs (make-constant :const 2)
+								:rhs (make-wire :var :a1))))
+		    (make-tuples :wires (list
+					 (make-wire :var :a0)
+					 (make-application :func :range
+						           :arguments (make-wire :var :a1))))))))
+(defun next-range (range x)
+          (make-application :func :next-range
+                                :arguments
+                                (list range x)))
```

Comment:
This does not belong in the spec file, should go to a different file. Likely `vampir.lisp` and we move all the `print-object` methods to `print.lisp` in the same folder

Reply:
I've tried the code movement you suggested, as well as rebasing this branch up to main.  Would you mind re-reviewing to see if we can get this in now?  Thanks!